clang/lib/Sema/SemaDeclCXX.cpp

```diff
@@ -18321,6 +18321,28 @@ bool Sema::CheckOverridingFunctionAttributes(const CXXMethodDecl *New,
     return true;
   }
 
+  // Virtual overrides: check for matching effects.
```

Comment:
Just a remark: Despite what I said wrt function declaration merging, I agree that it makes sense to check this and warn if overriding changes the effects of a function.